test/test_images/probe_helper/probe/cloudscheduler.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"context"
+	"fmt"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+const (
+	// CloudSchedulerSourceProbeEventType is the CloudEvent type of forward
+	// CloudSchedulerSource probes.
+	CloudSchedulerSourceProbeEventType = "cloudschedulersource-probe"
+)
+
+// CloudSchedulerSourceForwardProbe is the probe handler for forward probe requests
+// in the CloudSchedulerSource probe.
+type CloudSchedulerSourceForwardProbe struct {
```

Comment:
In the new design, cloud scheduler source probe will wait until the receiver receives a scheduler event, right? The drawback of this design is that the probe will need to wait for a long time to return (between 0-60s).

Reply:
That's right. On the other hand, with the previous design, we still had to wait some amount of time like 90s after creating the scheduler object in order to guarantee a tick. With this design, that wait is not necessary. So the total waiting time is identical in either case.

The biggest problem with a receiver-based scheduler probe is that the helper would receive ticks from every scheduler in the entire cluster, and it would need to keep the latest timestamps from all of those in memory in case they would be referenced in a later probe, without knowing whether or not they would ever be probed, or when to free that memory. The logic for handling that seems much more complicated than with this forward approach, and I stated, I believe the delay necessary in both cases to be similar.

Reply:
How about the delay for the static scheduler source?

Reply:
The current delay with the lone static scheduler is basically nil, the probe is instantaneous. With the new approach, I imagine it will average out to 30s. So yes, the delay for the static scheduler will increase substantially. I think this is an acceptable price to pay for cleaner unified code between the static and dynamic scheduler probes.

I did forget to address one subtlety, which is that with the increased delay, we need to worry about two scheduler probes happening at the same time for the static scheduler. I will try to fix this.

Reply:
This is now fixed with the receiver-based approach + liveness probe cleanup.